src/server/pps/cmds/cmds.go

```diff
@@ -1761,6 +1763,50 @@ func setClusterDefaults(ctx context.Context, pachctlCfg *pachctl.Config, r io.Re
 	return nil
 }
 
+// setProjectDefaults sets the project defaults to the result of reading from r.  Reprocess implies regenerate.
+func setProjectDefaults(ctx context.Context, pachctlCfg *pachctl.Config, r io.ReadCloser, regenerate, reprocess, dryRun bool) error {
```

Comment:
I think a cleaner signature would be to just pass this function the client object instead of ctx + config

Reply:
In some ways, but IMNSHO we need to migrate away from storing contexts in structs; they should always just be the first argument to a function which will need them, or need to pass them. Now that we use the internal client, maybe this quarter we can refactor its use of contexts.